lib/matplotlib/tests/test_axes.py

```diff
@@ -4863,6 +4863,25 @@ def test_pandas_indexing_hist():
     axes.hist(ser_2)
 
 
+def test_pandas_bar_align_center():
+    # Tests fix for issue 8767
+    pd = pytest.importorskip('pandas')
+
+    df = pd.DataFrame({'col1': [1] + [2],
+                       'col2': [1]*2,
+                       'col3': [1]*2})
+
+    fig, axArr = plt.subplots(2)
+
+    curr = 1
+    for ax in axArr:
+        rects = ax.bar(df.loc[df['col1'] == curr, 'col2'],
```

Comment:
I don't think it needs an image test either...but I do think just a not raises isn't great when testing a kwarg - like it could very well be doing something unexpected/not parsing the data correctly.